packages/uni_app/lib/view/common_widgets/connectivity_warning_card.dart

```diff
@@ -0,0 +1,78 @@
+import 'dart:async';
+
+import 'package:connectivity_plus/connectivity_plus.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+
+import '../../generated/l10n.dart';
+
+class ConnectivityWarning extends StatefulWidget {
+  const ConnectivityWarning({super.key});
+
+  @override
+  State<ConnectivityWarning> createState() => _ConnectivityWarningState();
+}
+
+class _ConnectivityWarningState extends State<ConnectivityWarning> {
+  bool isOffline = false;
+
+  StreamSubscription<ConnectivityResult>? connectivitySubscription;
+
+  @override
+  void initState() {
+    super.initState();
+    checkInternetConnection();
+  }
+
+  void checkInternetConnection() {
+    connectivitySubscription = Connectivity()
+        .onConnectivityChanged
+        .listen((result) {
+          setState(() {
+            isOffline = result == ConnectivityResult.none;
+          });
+    });
+
+    checkInitialConnection();
```

Comment:
Shouldn't this be done before the subscription?
If so, the code can be in the initState method

Reply:
Yes, you are right!